src/client/src/rt-components/tear-off/ExternalWindow.tsx

```diff
@@ -12,51 +11,46 @@ const defaultConfig: WindowConfig = {
 }
 
 export interface ExternalWindowProps {
-  title: string
-  onBlock: () => void
+  title?: string
+  onBlock?: () => void
   onUnload: () => void
-  config: WindowConfig
+  config?: WindowConfig
 }
 
-const defaultWindowProps: ExternalWindowProps = {
-  title: '',
-  onBlock: null as () => void,
-  onUnload: null as () => void,
-  config: defaultConfig,
-}
-
-class ExternalWindow extends React.Component<ExternalWindowProps & { platform: PlatformAdapter }> {
-  externalWindow: Window | null = null
-
-  async componentDidMount() {
-    const { config, platform } = this.props
+const ExternalWindow: FC<ExternalWindowProps> = ({
+  title = '',
+  onBlock = null as () => void,
+  onUnload = null as () => void,
+  config = defaultConfig,
+}) => {
+  const platform = usePlatform()
 
-    this.externalWindow = await platform.window.open(config, this.release)
+  let externalWindow: Window
```

Comment:
We may want to use the `useRef` hook here so that the variable is preserved across renders. As per [this article](https://reactjs.org/docs/hooks-faq.html#is-there-something-like-instance-variables) it's supposed to be the equivalent of instance fields in a class component